lib/search_files.js

```diff
@@ -0,0 +1,291 @@
+var fs = require('fs');
+var os = require('os');
+var path = require('path');
+var glob = require('glob');
+
+
+function isWindows() {
+  return process.platform === 'win32';
+}
+
+function isFunction(v) {
+  return (typeof v === 'function');
+}
+
+function isObject(v) {
+  return ({}.toString.call(v) === '[object Object]');
+}
+
+function isArray(v) {
+  return Array.isArray(v);
+}
+
+function isString(v) {
```

Comment:
I'd be fine with bringing lodash in for these utilities unless @tkellen has an objection